x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/insight/index.tsx

```diff
@@ -238,28 +158,48 @@ const InsightComponent = ({
       }),
     });
   }
-  const { data: alertData } = useContext(BasicAlertDataContext);
+  const { data: alertData, timestamp } = useContext(BasicAlertDataContext);
   const { dataProviders, filters } = useInsightDataProviders({
     providers: parsedProviders,
     alertData,
   });
-  const { totalCount, isQueryLoading, oldestTimestamp, hasError } = useInsightQuery({
-    dataProviders,
-    filters,
-  });
-  const timerange: TimeRange = useMemo(() => {
+  const relativeTimerange: TimeRange | null = useMemo(() => {
     if (relativeFrom && relativeTo) {
       const fromStr = relativeFrom;
       const toStr = relativeTo;
-      const from = parseDateWithDefault(fromStr, DEFAULT_FROM_MOMENT).toISOString();
-      const to = parseDateWithDefault(toStr, DEFAULT_TO_MOMENT, true).toISOString();
+      const alertRelativeDate = timestamp ? moment(timestamp) : moment();
+      const from = parseDateWithDefault(
+        fromStr,
+        DEFAULT_FROM_MOMENT,
+        false,
+        moment,
+        alertRelativeDate.toDate()
+      ).toISOString();
+      const to = parseDateWithDefault(
+        toStr,
+        DEFAULT_TO_MOMENT,
+        true,
+        moment,
+        alertRelativeDate.toDate()
+      ).toISOString();
       return {
-        kind: 'relative',
+        kind: 'absolute',
         from,
         to,
-        fromStr,
-        toStr,
       };
+    } else {
```

Comment:
super minor thing (feel free to ignore if you want) but I feel like an early return at the beginning of the function would make it easier to read